Make column reordering compatible with pinnable column (#3707)
hashicorp/design-system: .changeset/seven-heads-clean.md

---
"@hashicorp/design-system-components": minor
---

<!-- START components/table/advanced-table -->

> 

Reviewer: ```suggestion
```

> `AdvancedTable` - Added support for using pinnable and reorderable column simultaneously.

Reviewer: ```suggestion
`AdvancedTable` - Added support for allowing a pinnable first column and reorderable columns at the same time.
```

> 

Reviewer: ```suggestion
```
